modcc/functioninliner.cpp

```diff
@@ -1,85 +1,157 @@
 #include <iostream>
 
+#include "astmanip.hpp"
 #include "error.hpp"
 #include "functioninliner.hpp"
 #include "errorvisitor.hpp"
 
-expression_ptr inline_function_call(Expression* e)
+expression_ptr inline_function_call(const expression_ptr& e)
 {
-    if(auto f=e->is_function_call()) {
-        auto func = f->function();
-#ifdef LOGGING
-        std::cout << "inline_function_call for statement " << f->to_string()
-                  << " with body" << func->body()->to_string() << "\n";
-#endif
-        auto& body = func->body()->statements();
-        if(body.size() != 1) {
-            throw compiler_exception(
-                "can only inline functions with one statement", func->location()
-            );
+    auto assign_to_func = e->is_assignment();
+    auto ret_name = assign_to_func->lhs()->is_identifier()->clone();
```

Comment:
Does `ret_name` need to be cloned? Is it modified below, or just used as an input below?

Reply:
It's just used as an input, I will pass it as `Expression*`